v2/api/storage/v1alpha1api20210401/management_policy__status_arm_types_gen.go

```diff
@@ -134,14 +134,14 @@ type TagFilter_StatusARM struct {
 
 type DateAfterCreation_StatusARM struct {
 	//DaysAfterCreationGreaterThan: Value indicating the age in days after creation
-	DaysAfterCreationGreaterThan float64 `json:"daysAfterCreationGreaterThan,omitempty"`
+	DaysAfterCreationGreaterThan *float64 `json:"daysAfterCreationGreaterThan,omitempty"`
```

Comment:
See my comment about `MakeTypeOptional`

Reply:
Agreed; should only change the type, not optionality. This is happening because `AsPrimitiveType()` is throwing away layers of the composite type. See my other comment for how to use `TypeVisitor` to make the change while preserving all the layers.

Reply:
This is the behaviour I also noticed. So having pointer value is the right thing. When I added transformation using azure-arm.yaml spec by using `  target:  name: float`, it just replaced with a non pointer type. As of the changes made by @matthchr in #2126 this is right.

Reply:
Even with all the suggested changes, I'm getting the similar results for this one.

Reply:
That makes sense actually. Agreed that this is getting generated as expected. I was thinking it was right before but as you say actually it was not.